packages/react/src/products/hooks/useProductDetails.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * Hook to provide all kinds of data for the business logic attached to the product details.
+ *
+ * @module useProductDetails
+ * @category Products
+ * @subcategory Hooks
+ */
+import {
+  fetchProductDetails,
+  getAllProductSizesRemainingQuantity,
+  getProductBreadcrumbs,
+  getProductError,
+  getProductGroupedEntries,
+  isProductDuplicated,
+  isProductFetched,
+  isProductHydrated,
+  isProductLoading,
+} from '@farfetch/blackout-redux/products';
+import {
+  getProduct,
+  getProductLabelsByPriority,
+  getProductPromotions,
+  isProductOneSize,
+  isProductOutOfStock,
+} from '@farfetch/blackout-redux/entities';
+import { isProductInBag } from '@farfetch/blackout-redux/bags';
+import { useAction } from '../../helpers';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { StoreState } from '@farfetch/blackout-redux/src/types';
+import type { UseProductDetails } from './types';
+
+/**
+ * Provides state access for dealing with product details business logic.
+ * Also it fetches product details if there are none.
+ *
+ * @param {number} id - Product id to work on.
+ *
+ * @memberof module:products/hooks
+ * @returns {object} All the state needed to manage any product details information.
+ */
+const useProductDetails: UseProductDetails = id => {
+  const getProductDetails = useAction(fetchProductDetails);
```

Comment:
To not mistake this with the selector, what do you think of keeping the `fetchProductDetails` name?